ibis/expr/operations/relations.py

```diff
@@ -785,3 +785,29 @@ class NotExistsSubquery(Node):
 
     def output_type(self):
         return ir.ExistsExpr
+
+
+@public
+class FillNa(TableNode, sch.HasSchema):
+    """Fill null values in the table."""
+
+    table = Arg(rlz.table)
+    value = Arg(rlz.any)
```

Comment:
What happens with a `value` that doesn't match the type of one or more elements of `subset`?

Reply:
Perhaps this should accept a dictionary mapping column names to value to use for replacement?

Reply:
I think this is convenient,  but also a bit weird because now "subset" is leaked into "value" (they are redudant) so not sure

```
empty_values = {
    dt.int32: 0,
    dt.float64: 0.0,
    dt.string: ""
}

to_fill = {
    table[col].name: empty_values[table[col].type()] for col in table.columns
}
dt.fillna(value=to_fill)
```

Reply:
Does seem like pandas allows this? Seems reasonable to me then
```
df.fillna({'Name':'.', 'City':'.'}, inplace=True)
```

Reply:
Right, we don't need to support the full pandas API, but if there's a stricter subset (it sounds like there is) then we can definitely support that.

Definitely don't think we should map value type to value, and instead remove the `subset` argument and accept a mapping like @icexelloss is showing here. I'll throw out calling the argument `replacements` but no strong feeling there.

Reply:
So in this case the proposed usages would be:

Case 1: I want to fill all empty values with 0
```
table.fillna(0)
```

Case 2: I want to fill only a subset of columns with 0:
```
table.fillna({'A': 0, 'B': 0})
```

Case 3: I want to use different replacement values for different columns
```
table.fillna({'A': 1, 'B': 0})
```
I agree this is more concise than accepting both `value` and `subset`. And that `replacements` sounds better than `values`.

Reply:
> What happens with a `value` that doesn't match the type of one or more elements of `subset`?

I believe Pyspark will ignore in this case, whereas Pandas will actually fill (e.g. df.fillna({'float_col': 'string value'}) will do the replacement and promote the type of 'float_col' from float to object). So we may want to be more opinionated about the types matching like Pyspark is.

Reply:
I suppose there are a couple recipes that could be useful for folks coming from pandas as well:

```
# replace all na values in all columns with the same value
t.fillna({c: "." for c in t.columns})

# replace all na values in all columns of a given type with the same value
t.fillna({c: 0.0 for c, t in t.schema().items() if t == dt.float64})
```

Reply:
So for the first example, I think we are better off accepting either a scalar or a dict in the API, which would prevent users from having to type a dict comprehension in the case where they want to fill all columns with the same value.

For the second, I wonder if we want to be strict about something like 
```
t.fillna({'float_col': '.'})
```
giving the same behavior between Pandas/Dask and Pyspark. If so, we'd take care of that second example internally, i.e. in Pandas/Dask execution rules we'd compare the dtypes and only consider the columns with matching type.

Reply:
Actually I think Pyspark may only be specifically ignoring (str, numeric) mismatches, while it will cast ints and bools to floats and fillna with those values. It might be messy trying to converge this behavior internally